src/senaite/ast/browser/modals/rejectantibiotics.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.AST.
+#
+# SENAITE.AST is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free
+# Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2020-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+import itertools
+from datetime import datetime
+
+from bika.lims import api
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from senaite.ast import logger
+from senaite.ast import utils
+from senaite.ast.calc import update_sensitivity_result
+from senaite.ast.config import NOT_TESTED
+from senaite.ast.config import REPORT_EXTRAPOLATED_KEY
+from senaite.ast.config import REPORT_KEY
+from senaite.ast.config import RESISTANCE_KEY
+from senaite.core.api import dtime as dt
+from senaite.core.browser.modals import Modal
+
+
+class RejectAntibioticsModal(Modal):
+    """Modal that allows to reject antibiotics (flag them as Not Tested)
+    """
+
+    template = ViewPageTemplateFile("templates/reject_antibiotics.pt")
+
+    def __call__(self):
+        if self.request.form.get("submitted", False):
+            self.handle_submit()
+        return self.template()
+
+    @property
+    def analyses(self):
+        """Returns the analyses passed-in as UIDs through the request
+        """
+        analyses = map(api.get_object_by_uid, self.uids)
+        # be sure we rely on AST analyses only
+        analyses = filter(utils.is_ast_analysis, analyses)
+        # extend with siblings (AST-like analyses from same microorganism)
+        siblings = map(utils.get_ast_siblings, analyses)
+        siblings = list(itertools.chain.from_iterable(siblings))
+        analyses.extend(siblings)
+        # remove duplicates
+        analyses = set(analyses)
+        # exclude analyses meant for selective reporting
+        exclude = [REPORT_KEY, REPORT_EXTRAPOLATED_KEY]
+        analyses = filter(lambda an: an.getKeyword() not in exclude, analyses)
+        return analyses
+
+    @property
+    def antibiotics(self):
+        """Returns a list of antibiotics with the antibiotics assigned to
+        the analyses passed-in as UIDs through the request, sorted by name
+        ascending
+        """
+        kwargs = dict(filter_criteria=self.is_valid_antibiotic)
+        antibiotics = utils.get_antibiotics(self.analyses, **kwargs)
```

Comment:
You could pass in the method here directly instead of creating first a dictionary for it, e.g. `utils.get_antibiotics(self.analyses, filter_criteria=self.is_valid_antibiotic)` (but probably the line would then exceed 80 chars or?)